### ISSUE TITLE
[Crash Report] App becomes very unstable when song count exceeds a certain value

### ISSUE THREAD

**Issue by asciibene** [chr56/Phonograph_Plus#187]
### Crash log

```
Crash Report:

Package name:    player.phonograph.plus
App version:     1.6.0 (1060)
Git Commit Hash: 86916ee0b9f8df660b6b9792d273923588014126
Android version: 11 (SDK 30)
Device brand:    samsung  (by samsung)
Device model:    SM-A205W (code: a20)
Product name:    a20cs
Build version:   RP1A.200720.012.A205WVLSCCWD1 
                 (A205WVLSCCWD1)
Hardware:        exynos7884B
Language:        en


Stacktrace:
java.lang.ArrayIndexOutOfBoundsException: length=8068; index=-1
	at java.util.ArrayList.get(ArrayList.java:439)
	at ze.i.onViewAttachedToWindow(SourceFile:15)
	at androidx.recyclerview.widget.d.a(SourceFile:41)
	at androidx.recyclerview.widget.v0.b(SourceFile:220)
	at androidx.recyclerview.widget.GridLayoutManager.c1(SourceFile:222)
	at androidx.recyclerview.widget.LinearLayoutManager.O0(SourceFile:49)
	at androidx.recyclerview.widget.LinearLayoutManager.h0(SourceFile:809)
	at androidx.recyclerview.widget.GridLayoutManager.h0(SourceFile:45)
	at androidx.recyclerview.widget.RecyclerView.w(SourceFile:364)
	at androidx.recyclerview.widget.RecyclerView.v(SourceFile:64)
	at androidx.recyclerview.widget.RecyclerView.onLayout(SourceFile:8)
	at android.view.View.layout(View.java:24475)
	at android.view.ViewGroup.layout(ViewGroup.java:7383)
	at androidx.swiperefreshlayout.widget.SwipeRefreshLayout.onLayout(Unknown Source:58)
	at android.view.View.layout(View.java:24475)
	at android.view.ViewGroup.layout(ViewGroup.java:7383)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at android.view.View.layout(View.java:24475)
	at android.view.ViewGroup.layout(ViewGroup.java:7383)
	at androidx.coordinatorlayout.widget.CoordinatorLayout.r(SourceFile:392)
	at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(SourceFile:47)
	at android.view.View.layout(View.java:24475)
	at android.view.ViewGroup.layout(ViewGroup.java:7383)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at android.view.View.layout(View.java:24475)
	at android.view.ViewGroup.layout(ViewGroup.java:7383)
	at androidx.recyclerview.widget.v0.S(SourceFile:1)
	at androidx.recyclerview.widget.LinearLayoutManager.c1(SourceFile:249)
	at androidx.recyclerview.widget.LinearLayoutManager.O0(SourceFile:49)
	at androidx.recyclerview.widget.LinearLayoutManager.h0(SourceFile:809)
	at androidx.recyclerview.widget.RecyclerView.x(SourceFile:61)
	at androidx.recyclerview.widget.RecyclerView.v(SourceFile:72)
	at androidx.recyclerview.widget.RecyclerView.onLayout(SourceFile:8)
	at android.view.View.layout(View.java:24475)
	at android.view.ViewGroup.layout(ViewGroup.java:7383)
	at androidx.viewpager2.widget.ViewPager2.onLayout(Unknown Source:60)
	at android.view.View.layout(View.java:24475)
	at android.view.ViewGroup.layout(ViewGroup.java:7383)
	at v6.m.t(SourceFile:144)
	at v6.n.h(SourceFile:1)
	at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(SourceFile:41)
	at android.view.View.layout(View.java:24475)
	at android.view.ViewGroup.layout(ViewGroup.java:7383)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at android.view.View.layout(View.java:24475)
	at android.view.ViewGroup.layout(ViewGroup.java:7383)
	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
	at android.view.View.layout(View.java:24475)
	at android.view.ViewGroup.layout(ViewGroup.java:7383)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at android.view.View.layout(View.java:24475)
	at android.view.ViewGroup.layout(ViewGroup.java:7383)
	at com.sothree.slidinguppanel.SlidingUpPanelLayout.onLayout(SourceFile:142)
	at android.view.View.layout(View.java:24475)
	at android.view.ViewGroup.layout(ViewGroup.java:7383)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at android.view.View.layout(View.java:24475)
	at android.view.ViewGroup.layout(ViewGroup.java:7383)
	at androidx.drawerlayout.widget.DrawerLayout.onLayout(SourceFile:58)
	at android.view.View.layout(View.java:24475)
	at android.view.ViewGroup.layout(ViewGroup.java:7383)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at android.view.View.layout(View.java:24475)
	at android.view.ViewGroup.layout(ViewGroup.java:7383)
	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
	at android.view.View.layout(View.java:24475)
	at android.view.ViewGroup.layout(ViewGroup.java:7383)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at android.view.View.layout(View.java:24475)
	at android.view.ViewGroup.layout(ViewGroup.java:7383)
	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1829)
	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1673)
	at android.widget.LinearLayout.onLayout(LinearLayout.java:1582)
	at android.view.View.layout(View.java:24475)
	at android.view.ViewGroup.layout(ViewGroup.java:7383)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:332)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:270)
	at com.android.internal.policy.DecorView.onLayout(DecorView.java:1225)
	at android.view.View.layout(View.java:24475)
	at android.view.ViewGroup.layout(ViewGroup.java:7383)
	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:4260)
	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3695)
	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2618)
	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:9971)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1010)
	at android.view.Choreographer.doCallbacks(Choreographer.java:809)
	at android.view.Choreographer.doFrame(Choreographer.java:744)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:995)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:246)
	at android.app.ActivityThread.main(ActivityThread.java:8625)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)



```

Its been a problem since the original phonograph if my memory serves me well... its an old problem so its probably not caused by any recent change to the app. But dont take my word for it too seriously

Ive had this issue across several devices.

### Version

1.6.2

### Steps to reproduce the bug

Use app as intended... (Library -> 8087 files)
  Including such examples as:
    * _(attempting)_ to create playlists or playing queues
    * navigating the albums/artists/folders/etc tabs
    * using the search function through a local library of exactly 8087 songs.
    * properly displaying album cover arts (sometimes they appear, and some other day the app decides it will not want to display :( ) sometimes the cover art that gets displayed is not the right one or straight up a mess of random pixels (colored noise)
    * The screen controls become unresponsive and/or react incorrectly to input

Do note that these issues happen only with a large library with possibly many different files (on another device that has a lesser library of around 200 files I do not get any major problem worth reporting.)

### Environment

android 11 running on Galaxy A20

**Comment by chr56** [comment]
Sorry, no crash log, I am _blind_ for your situation.

**Comment by asciibene** [comment]
Okay i added a crash log (I easily managed to crash the app by selecting an album and pressing the back key)

**Comment by chr56** [comment]
Thanks. I have checked it.

#### 1. Issue from your crash log has already fixed

The crash log you post is on **version 1.6.0**, however, this crash has already fixed since **[v1.6.1](https://github.com/chr56/Phonograph_Plus/releases/tag/v1.6.1)**[^1].
Besides, the version you report, is **1.6.2**: so you reported a bug already fixed if you are using _v1.6.2_[^2], but you uploaded a crash log from legacy version! I am confusing!

Please upgrade all application cross all your devices.

#### 2. So, are there any other problems "making it very unstable"

Are you sure only this one?

#### 3. Send me all crash logs

If you,
- did not clean app cache
- not reinstalled app

Then, you _could_ acquire **all history crash log**:

All history crash log would be saved in app's _external cache directory_: `<Internal Storage Root>/Android/data/<Phonograph Plus Package Name>/cache`[^3], you can access it quickly access them through vanilla Document by `Main Drawer` > `More` > `View external files`.


[^1]: changelog v1.6.1: `Fix: possible random crash after selecting multiple items`, it was written like this but it fixed more that it describes.
[^2]: wake up, June is about coming, we are already in [v1.7.0](https://github.com/chr56/Phonograph_Plus/releases/tag/v1.7.0), _keep updating_, how difficult it is?
[^3]: package name: Stable channel `player.phonograph.plus`, Preview channel `player.phonograph.plus.preview`

**Comment by asciibene** [comment]
I realised I was using "stable channel" so I updated to 1.7.0 available on Preview channel and the app is now alot more useable.
